rotkehlchen/chain/arbitrum_one/modules/curve/decoder.py

```diff
@@ -29,4 +30,5 @@ def __init__(
             aave_pools=set(),
             curve_deposit_contracts={DEPOSIT_AND_STAKE_ZAP},
             curve_swap_routers={CURVE_SWAP_ROUTER_NG},
+            gauge_factory_address=CHILD_LIQUIDITY_GAUGE_FACTORY,
```

Comment:
If it's the same address for every chain, why add it as an argument? Just hardcode and not save it in the class. Only if it differs per chain would it make sense to have it per class.

Reply:
Not all the chains that we support have this factory contract and also it is not always the same since some chains have a different one but we don't support them now like fraxtal. One chain that doesn't has this contract is binance